src/index.ts

```diff
@@ -1,18 +1,50 @@
 import base64 = require('base-64');
+import queryStringHelpers = require('querystring');
+import urlHelpers = require('url');
+
+import { emptyStringToUndefined, getIndex, getKey, getValues, identity } from './helpers';
+import { mapMaybe, Maybe } from './maybe';
+import {
+  getQueryStringFromParsedUrl,
+  omitQueryParamFromUrl,
+  Query,
+  QueryOptions,
+  setQueryParamForUrl,
+} from './url-query';
 
 const TRACKING_PARAM = 'ixid';
 
-const trackingExpression = /(ixid=.*(?=&))|(ixid=.*)/i;
-export const _findTrackingParamsInUrl = (url: string) => {
-  const matches = url.match(trackingExpression);
+const getTrackingQueryParam = (query: Query): Maybe<string> =>
+  mapMaybe(x => (typeof x === 'string' ? x : getIndex(0, x)), getKey(TRACKING_PARAM, query));
```

Comment:
I'd avoid using a single-character variable, and replace it with `param` instead for clarity.

Reply:
Fixed! Thanks.